openshift-4.0/images/ose-etcd.yml

```diff
@@ -7,14 +7,18 @@ owners:
 
 content:
   source:
-    alias: ose-etcd
+    alias: etcd
     dockerfile: Dockerfile.rhel
```

Comment:
This file does not actually seem to exist in the upstream repo on master.

Reply:
> Should openshift/etcd with branch origin-4.0-etcd-v3.3.10 be the upstream? We need the Dockerfile to be in the source repo and branch.

@sosiouxme I think for full control this should be fine just to be clear on my process.

```bash
$ git remote -v
origin	git@github.com:openshift/etcd.git (fetch)
origin	git@github.com:openshift/etcd.git (push)
upstream	git@github.com:etcd-io/etcd.git (fetch)
upstream	git@github.com:etcd-io/etcd.git (push)
```
I fetched upstream and merge into origin master and pushed back to origin master repo making it 1:1 against upstream. I also pushed up all of the upstream tags.

```bash
git checkout `v3.3.10`
git checkout -b  origin-4.0-etcd-v3.3.10
[ commited work ]
git push --set-upstream origin origin-4.0-etcd-v3.3.10
```
So right now this branch is current with etcd release v3.3.10 + Dockerfile.rhel. This would not be possible with `git@github.com:etcd-io/etcd.git`

Can I change the above ocp-build-data/openshift-4.0/group.yml to reflect?

```yaml
 etcd: 
   url: git@github.com:openshift/etcd.git 
   branch: 
     target: origin-4.0-etcd-v{MAJOR}.{MINOR}.{PATCH}
     fallback: master <-- we don't want this can we be 100% explicit on target?
```

Reply:
```
commit 4dd75379e10475b1a6d18e72196218a0165082d1 (HEAD -> origin-4.0-etcd-v3.3.10, origin/origin-4.0-etcd-v3.3.10)
Author: Sam Batschelet <sbatsche@redhat.com>
Date:   Thu Nov 29 11:39:48 2018 -0500

    version: origin-4.0-etcd-v3.3.10
    
    Signed-off-by: Sam Batschelet <sbatsche@redhat.com>

commit 27fc7e2296f506182f58ce846e48f36b34fe6842 (tag: v3.3.10)
Author: Gyuho Lee <leegyuho@amazon.com>
Date:   Wed Oct 10 10:17:54 2018 -0700

    version: 3.3.10
    
    Signed-off-by: Gyuho Lee <leegyuho@amazon.com>
```

Reply:
Yes to everything you said. The fallback strategy is just because most people are doing 4.0 work on master, and we want to use that for now and automatically start using the release branch for 4.0 when that gets created. You are already on a release branch, so no need.

The reason for the {MAJOR}{.{MINOR} business is so that when we branch for 4.1 we can be lazy and not have to change anything except the top of the group file. I'm betting that origin-4.1 will not use etcd v3.3.10 so this will break at that point and we'll have to determine what you want. If there's anything more future-proof you can put here, let's do that.

Reply:
I like that everything is explicit but I am open to change if you feel this will become a blocker.